src/internal/connector/exchange/container_resolver.go

```diff
@@ -74,13 +76,21 @@ func (cr *containerResolver) idToPath(
 		return p, nil
 	}
 
-	parentPath, err := cr.idToPath(ctx, *c.GetParentFolderId(), depth+1)
+	parentPath, err := cr.idToPath(ctx, *c.GetParentFolderId(), depth+1, useIDInPath)
 	if err != nil {
 		return nil, errors.Wrap(err, "retrieving parent folder")
 	}
 
-	fullPath := parentPath.Append(*c.GetDisplayName())
+	toAppend := *c.GetDisplayName()
+	if useIDInPath {
+		toAppend = *c.GetId()
+	}
+
+	fullPath := parentPath.Append(toAppend)
+	locPath := parentPath.Append(*c.GetDisplayName())
```

Comment:
we're going to need to return the locPath as well and then use that in place of `parentPath` here. Otherwise we'll end up with locPaths that have mixed IDs/display names

Reply:
ooh, yup, good catch